--- a/src/hikrobot_camera.cpp
+++ b/src/hikrobot_camera.cpp
@@ -59,6 +59,7 @@
     image_transport::CameraPublisher record_image_pub = main_cam_image.advertiseCamera(record_image_topic, 2);
     ros::Publisher preview_pub = hikrobot_camera.advertise<sensor_msgs::CompressedImage>(preview_compressed_topic, 2);
     ros::Publisher frame_info_pub = hikrobot_camera.advertise<std_msgs::String>("/hikrobot_camera/frame_info", 30);
+    ros::Publisher record_frame_info_pub = hikrobot_camera.advertise<std_msgs::String>("/hikrobot_camera/record_frame_info", 10);
 
     sensor_msgs::Image image_msg;
     sensor_msgs::CameraInfo camera_info_msg;
@@ -155,6 +156,9 @@
                 image_msg.header.frame_id = "hikrobot_camera";
                 camera_info_msg.header = image_msg.header;
                 record_image_pub.publish(image_msg, camera_info_msg);
+                std_msgs::String record_frame_info_msg;
+                record_frame_info_msg.data = std::to_string(frame_meta.frame_num);
+                record_frame_info_pub.publish(record_frame_info_msg);
                 if (next_record_stamp <= 0.0)
                 {
                     next_record_stamp = frame_time + record_period;
